pennylane/templates/embeddings.py

```diff
@@ -0,0 +1,266 @@
+
+.. currentmodule:: pennylane.templates.embeddings
+
+This module provides quantum circuit architectures that can embed features into a quantum state.
+The features are associated with gate parameters (see also Schuld & Killoran 2019
+:cite:`schuld2018quantum`).
+
+Qubit architectures
+-------------------
+
+Angle embedding
+***************
+
+Encodes ``n`` features into the rotation angles of ``n`` qubits. The rotation can be Pauli-X, -Y or -Z.
+
+.. autosummary::
+
+    AngleEmbedding
+
+Amplitude embedding
+*******************
+
+Encodes :math:`2^n` features into the amplitude vector of :math:`n` qubits. The absolute square of all features
+has to add up to one.
+
+.. note::
+
+    This embedding is a wrapper for PennyLane`s :mod:`QubitStateVector()`, and only works with backends which
+    implement this method.
+
+.. autosummary::
+
+    AmplitudeEmbedding
+
+Basis embedding
+***************
+
+Encodes :math:`n` bits into a basis state of :math:`n` qubits.
+
+.. note::
+
+    This embedding is a wrapper for PennyLane`s :mod:`BasisState()`, and only works with backends which
+    implement this method.
+
+.. autosummary::
+
+    BasisEmbedding
+
+Continuous-variable architectures
+---------------------------------
+
+Squeezing embedding
+*******************
+
+Encodes :math:`M` features into the squeezing amplitude or phase of :math:`M` modes.
+
+
+.. autosummary::
+
+    SqueezingEmbedding
+
+
+Displacement embedding
+**********************
+
+Encodes :math:`M` features into the displacement amplitude or phase of :math:`M` modes.
+
+.. autosummary::
+
+    DisplacementEmbedding
+
+Code details
+^^^^^^^^^^^^
+"""
+#pylint: disable-msg=too-many-branches,too-many-arguments,protected-access
+from pennylane import RX, RY, RZ, BasisState, Squeezing, Displacement, QubitStateVector
+from collections.abc import Iterable
+
+
+def AngleEmbedding(features, wires, rotation='X'):
+    r"""
+    Uses the entries of ``features`` as rotation angles of qubits.
+
+    The details of the strategy are defined by the `rotation` parameter:
+     * ``rotation = 'X'`` uses the features to chronologically apply Pauli-X rotations to qubits
+     * ``rotation = 'Y'`` uses the features to chronologically apply Pauli-Y rotations to qubits
+     * ``rotation = 'Z'`` uses the features to chronologically apply Pauli-Z rotations to qubits
+
+    The length of ``features`` has to be smaller or equal to the number of qubits. If there are fewer entries in
+    ``features`` than rotations, the circuit does not apply the remaining rotation gates.
+
+    This embedding method can also be used to encode a binary sequence into a basis state. Choose ``rotation='X'``
+    and features of a nonzero value of :math:`\pi /2` only where a qubit has to be prepared in state 1.
+
+    Args:
+        features (array): Input array of shape ``(N, )``, where N is the number of input features to embed
+        wires (int): List of qubit indices for the qubits used for the embedding
```

Comment:
should this be wires(Sequence[int] ) rather than wires(int) to get the list of wires we want the rotation to act on?

Reply:
Thanks, changed this!